### PR TITLE
Replace Icon static properties with bona fide enum cases

### PR DESCRIPTION
**Pull request by dwroth**
TL;DR - After this change goes out, the format for referencing Icon images will change from `Icon.theIconName` to `Icon.theIconName.image`

Currently, we the Icon enum doesn't have any actual cases; it's just a namespace.  This works fine, but doesn't allow us some of the power that enums provide.  These Icons are not iterable (for tests), nor is it possible to hold onto a generic Icon object (only the UIImage that it returns).  This change replaces the static image properties with string backed enum cases.  These strings map to the actual names of the underlying images.  Additionally, I added a property to the enum to return the image with the corresponding name.

### CODE REVIEW

File: Thumbprint/Targets/Playground/Classes/Inspectable/InspectableViews/IconButton+InspectableView.swift [hunk 1]
```diff
@@ -18,7 +18,7 @@ extension IconButton: InspectableView {
     }
 
     static func makeInspectable() -> UIView & InspectableView {
-        let button = IconButton(icon: Icon.contentActionsFilterMedium, accessibilityLabel: "Demo icon button")
+        let button = IconButton(icon: Icon.contentActionsFilterMedium.image, accessibilityLabel: "Demo icon button")
```
Comment (scottasoutherland) [inline Thumbprint/Targets/Playground/Classes/Inspectable/InspectableViews/IconButton+InspectableView.swift:21]: not neccesarily something you need to address as it's a personal pet peeve, but it bothers me when things have a parameter called "icon" but the then they don't take an "icon" but an "image" 😬

**Review by scottasoutherland — APPROVED** [review]
